File: Javascript/String/Determine-If-String-Halves-Are-Alike.js
/**
 * @param {string} s
 * @return {boolean}
 */

const halvesAreAlike = (s) => {
  let middle = s.length / 2;

  return countVowels(s.slice(0, middle)) == countVowels(s.slice(middle));
};

const countVowels = (s) => {
  let set = new Set(["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]);
  let count = 0;

  for (letter of s) {
    if (set.has(letter)) {
      count++;
    }
  }

  return count;
};

File: Go/String/Determine-If-String-Halves-Are-Alike.go
package main

func main() {

}

func halvesAreAlike(s string) bool {
	middle := len(s) / 2

	return countVowels(s[:middle]) == countVowels(s[middle:])
}

func countVowels(s string) int {
	vowels := []rune{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}

	count := 0

	for _, letter := range s {
		if contains(vowels, letter) {
			count++
		}
	}

	return count
}

func contains(s []rune, ch rune) bool {
	for _, vowel := range s {
		if vowel == ch {
			return true
		}
	}

	return false
}